# ¿Cómo medir rendimiento y riesgo en un portafolio? I

<img style="float: right; margin: 0px 0px 15px 15px;" src="http://www.picpedia.org/clipboard/images/stock-portfolio.jpg" width="600px" height="400px" />

> En el módulo anterior adquirimos técnicas analíticas (con base en teoría de probabilidad) para describir la distribución de los rendimientos de un activo.

> En este módulo adquiriremos herramientas analíticas para describir la distribución de rendimientos de un portafolio de activos financieros.

> Antes de formular el problema de selección óptima de portafolios, necesitamos tener medidas de rendimiento y riesgo del portafolio.

> La idea principal que nos debemos llevar de este módulo es que el riesgo de un portafolio no se obtiene simplemente sumando los riesgos individuales de cada activo que conforma el portafolio. Por el contrario, un portafolio puede tener menos riesgo que los instrumentos mismos: **diversificación**.

**Objetivos:**
- Medir rendimiento esperado en un portafolio a partir de los rendimientos esperados de los activos que lo conforman.
- Medir el riesgo en un portafolio.

*Referencias*
- Notas del curso "Portfolio Selection and Risk Management", Rice University, disponible en Coursera.
___

## 1. Midiendo rendimientos esperados de un portafolio

- Entonces, ya sabemos obtener el rendimiento esperado de un único activo:
 - dada la distribución de probabilidad de los posibles escenarios económicos, o
 - dada una serie de tiempo de rendimientos (precios) pasados.
- De manera similar, aprendimos a medir el riesgo en un único activo.
- Raramente invertimos en un único activo; típicamente, invertimos en una colección de diferentes activos: invertimos en un **portafolio**.
- Entonces, necesitamos desarrollar medidas similares de rendimiento y riesgo para un portafolio de activos.

### 1.1. Recordamos para un solo activo
- Dada la distribución de rendimientos, usamos la tendecia central (media).
- El rendimiento esperado es entonces el promedio ponderado por probabilidad de los rendimientos posibles.

**Ejemplo.** Supongamos que tenemos inversión en activos de Toyota, Walmart y Pfizer. Tenemos cuatro posibles estados económicos:

In [1]:
import numpy as np
import pandas as pd

In [2]:
# Creamos tabla
tabla = pd.DataFrame(columns=['Prob', 'Toyota', 'Walmart', 'Pfizer'], index=['Expansion', 'Normal', 'Recesion', 'Depresion'])
tabla.index.name = 'State'
tabla['Prob']=np.array([0.1, 0.4, 0.3, 0.2])
tabla['Toyota']=np.array([0.06, 0.075, 0.02, -0.03])
tabla['Walmart']=np.array([0.045, 0.055, 0.04, -0.01])
tabla['Pfizer']=np.array([0.025, -0.005, 0.01, 0.13])

tabla.round(6)

Prob  Toyota  Walmart  Pfizer
State                                   
Expansion   0.1   0.060    0.045   0.025
Normal      0.4   0.075    0.055  -0.005
Recesion    0.3   0.020    0.040   0.010
Depresion   0.2  -0.030   -0.010   0.130

Calcular el rendimiento esperado para cada compañía.

In [3]:
# Toyota
ErT = (tabla['Prob'] * tabla['Toyota']).sum()
ErT

0.036

In [4]:
# Walmart
ErW = (tabla['Prob'] * tabla['Walmart']).sum()
ErW

0.036500000000000005

In [5]:
# Pfizer
ErP = (tabla['Prob'] * tabla['Pfizer']).sum()
ErP

0.029500000000000002

### 1.2. ¿Y si tenemos un portafolio de activos financieros?

> Un **portafolio** es una colección de diferentes activos junto con una ponderación relativa a cada activo. De esta manera, el rendimiento de un portafolio se trata como el rendimiento de un solo activo.

Entonces, si $r_i$ y $w_i$ son el rendimiento y la ponderación del activo $i$ ($i=1,2,\dots,n$), respectivamente, se tiene que el rendimiento del portafolio $r_p$ es:

$$r_p=\sum_{i=1}^{n}w_ir_i$$

**Ejemplo.** Consideramos un portafolio conformado por solamente dos activos de la tabla anterior. La ponderación relativa a cada activo es $0.5$ Toyota y $0.5$ Pfizer.

Revisemos el rendimiento de este portafolio en cada estado económico.

In [6]:
# Expansión
0.5 * 0.060 + 0.5 * 0.025

0.042499999999999996

In [7]:
# Normal
0.5 * 0.075 + 0.5 * -0.005

0.034999999999999996

In [8]:
# Calculamos rendimiento del portafolio en cada estado
tabla['PortTP'] = 0.5 * tabla['Toyota'] + 0.5 * tabla['Pfizer']
tabla

Prob  Toyota  Walmart  Pfizer  PortTP
State                                           
Expansion   0.1   0.060    0.045   0.025  0.0425
Normal      0.4   0.075    0.055  -0.005  0.0350
Recesion    0.3   0.020    0.040   0.010  0.0150
Depresion   0.2  -0.030   -0.010   0.130  0.0500

Ya que tenemos los posibles rendimientos del portafolio junto con las probabilidades asociadas, podemos calcular el rendimiento esperado del portafolio (tal cual lo hacíamos para un solo activo):

In [9]:
# Calculamos rendimiento esperado del portafolio
ErTP = (tabla['Prob'] * tabla['PortTP']).sum()
ErTP

0.03275

In [10]:
# Recordamos rendimientos esperados individuales
ErT, ErP

(0.036, 0.029500000000000002)

In [11]:
##### Rendimiento esperado ponderado
0.5 * ErT + 0.5 * ErP

0.03275

**Actividad.** Consideramos un portafolio conformado por solamente dos activos de la tabla anterior. La ponderación relativa a cada activo es $0.5$ Walmart y $0.5$ Pfizer.
- Calcular el rendimiento de este portafolio en cada estado económico.
- Calcular el rendimiento esperado del portafolio.

In [12]:
# Calculamos rendimiento del portafolio en cada estado
tabla['PortWP'] = 0.5 * tabla['Walmart'] + 0.5 * tabla['Pfizer']

In [13]:
# Mostramos
tabla

Prob  Toyota  Walmart  Pfizer  PortTP  PortWP
State                                                   
Expansion   0.1   0.060    0.045   0.025  0.0425   0.035
Normal      0.4   0.075    0.055  -0.005  0.0350   0.025
Recesion    0.3   0.020    0.040   0.010  0.0150   0.025
Depresion   0.2  -0.030   -0.010   0.130  0.0500   0.060

In [14]:
# Calculamos rendimiento esperado del portafolio
ErWP = (tabla['Prob'] * tabla['PortWP']).sum()
ErWP

0.033

In [15]:
# Recordamos rendimientos esperados individuales
ErW, ErP

(0.036500000000000005, 0.029500000000000002)

In [16]:
# Rendimiento esperado ponderado
0.5 * ErW + 0.5 * ErP

0.033

Alternativamente, no es difícil notar en este punto, que podríamos haber encontrado el rendimiento esperado del portafolio usando los rendimientos esperados de cada uno de los activos.

En general, dado que $r_p=\sum_{i=1}^{n}w_ir_i$, se tiene:

$$E[r_p]=E\left[\sum_{i=1}^{n}w_ir_i\right]=\sum_{i=1}^{n}w_iE\left[r_i\right],$$

pues la esperanza es un operador lineal.

**Actividad.** Consideramos un portafolio conformado por los tres activos de la tabla anterior, e igualmente ponderado en cada activo.
- Calcular el rendimiento esperado del portafolio.

In [17]:
# Calcular
ErTWP = 1 / 3 * ErT + 1 / 3 * ErW + 1 / 3 * ErP
ErTWP

0.034

**Conclusión.** El rendimiento esperado de un portafolio es simplemente el promedio ponderado (con las ponderaciones del portafolio) del rendimiento esperado de los activos individuales.
___

## 2. Midiendo el riesgo en un portafolio

### 2.1. Recordamos para un solo activo
- Para un solo activo usamos la dispersión, medida con la desviación estándar de la distribución de rendimientos.
- La varianza es el promedio ponderado por probabilidad de las desviaciones al cuadrado de la media.

Aprendimos que el rendimiento esperado de un portafolio es el promedio ponderado del rendimiento esperado de los activos individuales. Este no es el caso para el riesgo.

**Ejemplo.** Retomamos el ejemplo de Toyota, Walmart y Pfizer.

In [18]:
# Creamos tabla
tabla

Prob  Toyota  Walmart  Pfizer  PortTP  PortWP
State                                                   
Expansion   0.1   0.060    0.045   0.025  0.0425   0.035
Normal      0.4   0.075    0.055  -0.005  0.0350   0.025
Recesion    0.3   0.020    0.040   0.010  0.0150   0.025
Depresion   0.2  -0.030   -0.010   0.130  0.0500   0.060

Calcular la volatilidad para cada compañía

In [19]:
# Toyota
sT = (tabla['Prob'] * (tabla['Toyota'] - ErT)**2).sum()**0.5
sT

0.04017461885320133

In [20]:
# Walmart
sW = (tabla['Prob'] * (tabla['Walmart'] - ErW)**2).sum()**0.5
sW

0.02408837894089181

In [21]:
# Pfizer
sP = (tabla['Prob'] * (tabla['Pfizer'] - ErP)**2).sum()**0.5
sP

0.05111017511220246

### 2.2. ¿Y si tenemos un portafolio de activos financieros?

**Ejemplo.** Consideramos, de nuevo, un portafolio $0.5$ Toyota y $0.5$ Pfizer.

In [22]:
# Inclusión del portafolio
tabla

Prob  Toyota  Walmart  Pfizer  PortTP  PortWP
State                                                   
Expansion   0.1   0.060    0.045   0.025  0.0425   0.035
Normal      0.4   0.075    0.055  -0.005  0.0350   0.025
Recesion    0.3   0.020    0.040   0.010  0.0150   0.025
Depresion   0.2  -0.030   -0.010   0.130  0.0500   0.060

Encontrar la volatilidad del portafolio.

In [23]:
# Volatilidad
sTP = (tabla['Prob'] * (tabla['PortTP'] - ErTP)**2).sum()**0.5
sTP

0.012867109232457772

In [24]:
# Volatilidades individuales
sT, sP

(0.04017461885320133, 0.05111017511220246)

In [25]:
# Volatilidad ponderada
0.5 * sT + 0.5 * sP

0.045642396982701894

#### ¿Qué pasó?
- Combinamos con igual proporción un activo con volatilidad del 4.02% con otro de 5.11%,
- y obtuvimos un portafolio con volatilidad de 1.29%.
- Claramente, 
$$1.29\%\ll0.5\times 4.02\%+0.5\times 5.11\%$$

**Conclusión.** El riesgo de un portafolio no es el promedio ponderado de los riesgos individuales.
___

# Anuncios parroquiales

## 1. Les dejo este enlace de un [artículo interesante](https://www.economist.com/buttonwoods-notebook/2016/01/13/stocks-for-the-long-run): los activos no siempre son mejor apuesta que los instrumentos de deuda, aunque la teoría nos diga lo contrario.
## 2. Pueden ver las calificaciones en este [enlace](https://docs.google.com/spreadsheets/d/1BwI1Mm7B3xxJ-jQIQEDQ_WdRHyehZrQBpHGd0hY9fU4/edit?usp=sharing)

<script>
  $(document).ready(function(){
    $('div.prompt').hide();
    $('div.back-to-top').hide();
    $('nav#menubar').hide();
    $('.breadcrumb').hide();
    $('.hidden-print').hide();
  });
</script>

<footer id="attribution" style="float:right; color:#808080; background:#fff;">
Created with Jupyter by Esteban Jiménez Rodríguez.
</footer>